.github/workflows/build-publish-docker.yml

```diff
@@ -105,6 +105,24 @@ jobs:
         id: image_output
         run: echo "image_tag=${{ steps.meta.outputs.version }}" >> "$GITHUB_OUTPUT"
 
+  unit-test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    needs: build-and-push-docker-image
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm i
+
+      - name: Run Vitest tests
+        run: npm run test:unit
```

Comment:
Can we move this job to be the first job in the workflow? The other jobs should depend on the unit tests.

Imo the best order would be;

1. Unit tests
2. Build docker images
3. End-to-end tests

Reply:
Yes, done!